### ISSUE TITLE
alt text on image embeds should be imported and rendered in Frenklin

### ISSUE THREAD

**Issue by hsaginor** [headwirecom/helix-sportsmagazine#13]
Currently images are imported without alt text. We should also investigate how authors would add alt text in google docs.

**Comment by hsaginor** [comment]
@icaraps How should we store this in the google document?

**Comment by icaraps** [comment]
@hsaginor  Alt text is supported in google doc ... Isn't the alt text from the HTML retained when you copy paste it to google doc ?

**Comment by hsaginor** [comment]
@hsaginor There is no way to import directly into google doc. The importer saves everything as word documents and then I manually upload them. I am not sure how we transfer this information yet.

**Comment by hsaginor** [comment]
@icaraps My guess is that this is something that could be fixed in the [Helix Importer](https://github.com/adobe/helix-importer). Our transformation script just selects the markup we want to import and returns that. But even if the img tags we are returning contain alt attribute the alt text doesn't get preserved even though there is a way to pass it (as image title) in markdown - https://www.markdownguide.org/basic-syntax/#images-1

Here is an example importing a page with imageEmbed - https://www.golfdigest.com/content/golfdigest-com/en/golf-news-tours/_default/article/2010/1/justintimberlake.html
The imported img tag is `<img class="photo-gallery-launch a-Image" src="//golfdigest.sports.sndimg.com/content/dam/images/golfdigest/fullset/2015/08/14/55ce4f06b91019d74c98ef80_Justin-Timberlake-Interview-Staff.jpg.rend.hgtvcom.966.690.suffix/1573513695924.jpeg" style="object-position: 50% 10%; background-image: none;" loading="lazy" alt="Justin-Timberlake-Interview-Staff.jpg">`
Markdown created by importer from this is `![Justin-Timberlake-Interview-Staff.jpg][image0]`

From there I think it should also be preserved when the document is saved as MS Word and later converted to Google Doc when we upload it to gdrive. But assuming alt text is not lost in those steps at the moment it looks like the importer is not handling it through markdown.

**Comment by hsaginor** [comment]
@icaraps Actually you can ignore my last comment. It looks like alt text is actually imported if you check that document - https://docs.google.com/document/d/19PKC1k5_RHivYuD5J2_SgBr42CFMV4cxH9Z0csJDLdg/edit

![Screen Shot 2023-09-06 at 10 25 11 PM](https://github.com/headwirecom/helix-sportsmagazine/assets/1476638/b78c0405-2826-4900-b91e-436ecb7ac3b6)

**Comment by hsaginor** [comment]
I should make sure the import script adds alt text to all images. But do you have any examples where it's not imported?

**Comment by hsaginor** [comment]
@icaraps I see. The best I have seen on golfdigest do for these images is actually use the file name which is slightly better. If we reimport this page it will use the very cryptic "1491590551" as alt text. This is what the data is. The question is should we really force the importer to do better than the data it receives if we can't even apply a consistent rule? I see this as an editorial job.

**Comment by icaraps** [comment]
> I see this as an editorial job.

You're right, this would be the ideal world where editors define the right alt data for their images. I was hoping we could find a fallback solution that of course wouldn't be perfect but still better than the non human readable alt ...

**Comment by hsaginor** [comment]
@icaraps Well we can set it to file name if alt attribute is not set, empty or is a number like in this case. Leave it unchanged otherwise.

**Comment by hsaginor** [comment]
@icaraps See re-imported document with alt text set to file name https://docs.google.com/document/d/1RIMNgGL_zs11IPH7BXwuzba-Z1yCvG45roxTWT6kRg4/edit

But I don't see alt text being rendered by Franklin when I preview the page https://main--helix-sportsmagazine--headwirecom.hlx.page/content-v2/golf-news-tours/_default/article/2023/7/how-historic-would-scottie-scheffler-pga-tour-season-be-with-average-putting-stats
That wouldn't me in import issue.

**Comment by icaraps** [comment]
@hsaginor indeed it looks like a bug on hlx side because it's still documented https://www.hlx.live/docs/authoring#images-and-videos. I'll ask on their channel ...
